### PR TITLE
updated module that dumps raw waveforms for ML training

### PR DESCRIPTION
**Pull request by mhlswang**
added a new parameter to set the total number of raw waveforms to dump per event to help spread sampling of waveforms across events more evenly.  It also randomizes the list of mc tracks whose raw waveforms are dumped to further reduce possible correlations.

### CODE REVIEW

**Comment by knoepfel** [comment]
Looks fine to me @mhlswang.  Will approve once we have L2 approval.

**Comment by knoepfel** [comment]
Go ahead and add it, @mhlswang

File: larrecodnn/ImagePatternAlgs/Modules/NoiseWaveformDump_module.cc [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  }
+
+  std::cout << " !!!!! size of noisechannels: " << noisechannels.size() << std::endl;
+  if (!noisechannels.size()) return;
+
+  //save noise
+  int noisechancount = 0;
+
+  // .. create a vector for shuffling the wire channel indices
+  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```
Comment (knoepfel) [inline larrecodnn/ImagePatternAlgs/Modules/NoiseWaveformDump_module.cc:273]: I guess you don't require reproducibility for this...?

Comment (mhlswang) [reply larrecodnn/ImagePatternAlgs/Modules/NoiseWaveformDump_module.cc:269]: no I don't

**Comment by mhlswang** [comment]
@knoepfel I think I implemented all of your suggestions, please take a look.

**Comment by knoepfel** [comment]
trigger build

**Comment by knoepfel** [comment]
@mhlswang, thanks for addressing the changes so promptly.  Looks good!  Assuming the CI comes back successfully, I think we're good to go.

**Comment by yangtj207** [comment]
+1

**Comment by lgarren** [comment]
approve

**Pull request merged**